configure.ac

```diff
@@ -959,7 +959,7 @@ AS_IF([test x"$enable_shared" != "xno"],
     [*-*-cygwin*],
       [mksharedlib='$(FLEXLINK)'
       mkmaindll='$(FLEXLINK) -maindll'
-      shared_libraries_supported=true],
+      shared_libraries_supported=$with_sharedlibs],
```

Comment:
This utilises the result determined on L785 (just like the mingw and msvc cases) - `$with_sharedlibs` will have been correctly inferred as `false` here (because `flexdll.h` and `flexlink` were not found), but then that result was ignored here.